src/parseWithPointers.ts

```diff
@@ -12,30 +22,410 @@ export const parseWithPointers: SourceMapParser = <T>(value: string): IParserRes
 
   if (!value || !value.trim().length) return parsed;
 
-  const result = jsonSourceMap.parse(value);
-  parsed.data = result.data;
-  parsed.pointers = transformPointers(result.pointers);
+  const validations: IValidationResult[] = [];
 
-  return parsed;
+  return {
+    ...parse(value, validations, {
+      disallowComments: true,
+    }),
+    validations,
+  };
 };
 
-const transformPointers = (pointers: any) => {
-  const transformed: IParserResultPointers = {};
+// based on jsonc-parser source code
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+interface IJSONVisitor extends JSONVisitor {
+  onNewLine?: (lineNumber: ParseErrorCode, offset: number, length: number) => void;
+}
 
-  for (const path in pointers) {
-    if (!pointers.hasOwnProperty(path)) continue;
+function toPointer(fragments: Segment[]) {
+  return ['', ...fragments.filter(item => item !== '')].join('/');
+}
 
-    transformed[path] = {
+/**
+ * Parses the given text and returns the object the JSON content represents. On invalid input, the parser tries to be as fault tolerant as possible, but still return a result.
+ * Therefore always check the errors list to find out if the input was valid.
+ */
+function parse(text: string, errors: IValidationResult[] = [], options?: ParseOptions): any {
+  let currentProperty: string | null = null;
+  let currentParent: any = [];
+  let currentLineNumber = 1;
+  const offsets: number[] = [];
+  const previousParents: any[] = [];
+  const previousPointers: JSONPath = [];
+  const pointers = {
+    '': {
       start: {
-        // json-source-map is zero-based
-        line: pointers[path].value.line + 1,
+        line: 1,
       },
       end: {
-        // json-source-map is zero-based
-        line: pointers[path].valueEnd.line + 1,
+        line: 0,
       },
-    };
+    },
+  };
+
+  function onValue(value: any) {
+    if (Array.isArray(currentParent)) {
+      const index = (currentParent as any[]).push(value) - 1;
+
+      if (previousParents.length > 0) {
+        pointers[toPointer([...previousPointers.slice(0, previousPointers.length - 1), index])] = {
```

Comment:
Not mandatory, but you can do

```suggestion
        pointers[toPointer(previousPointers.slice(0, previousPointers.length - 1).concat(index])] = {
```

Reply:
:trollface: